### PR TITLE
[Enhancement] Bump FE/BE's hadoop to 3.4.0

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## Why I'm doing:
For the CVE problem, we need to upgrade Hadoop SDK from 3.3.6 -> 3.4.0
It will introduce aws java SDK v2, so we can delete SDK v1.

## What I'm doing:
Do it.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by imay — COMMENTED** [review]
it seems to me that you just change the AWS client dependency, can you rename this PR's title?

File: fe/fe-core/pom.xml [hunk 1]
```diff
@@ -317,24 +316,6 @@ under the License.
             </exclusions>
         </dependency>
 
-        <dependency>
-            <groupId>software.amazon.awssdk</groupId>
-            <artifactId>bundle</artifactId>
-            <version>${awsv2.version}</version>
```
Comment (stephen-shelby) [inline fe/fe-core/pom.xml:323]: exclude unused dependency

**Comment by Smith-Cruise** [comment]
> it seems to me that you just change the AWS client dependency, can you rename this PR's title?

aws SDK v2 is introduced by hadoop-aws : 3.4.0

Because we upgraded Hadoop, we also upgraded aws.

File: fe/fe-core/pom.xml (hunk 1 above)
Comment (Smith-Cruise) [reply fe/fe-core/pom.xml:323]: hard do it, hadoop-aws will include `bundle` automatically.

File: java-extensions/jni-runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.starrocks</groupId>
+        <artifactId>java-extensions</artifactId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>jni-runtime</artifactId>
```
Comment (stephen-shelby) [inline java-extensions/jni-runtime/pom.xml:12]: common-runtime

**Review by stephen-shelby — COMMENTED** [review]
good job!

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java [hunk 3]
```diff
@@ -315,19 +313,6 @@ private static CommandLineOptions parseArgs(String[] args) {
...
 
-    // To resolve: "Multiple HTTP implementations were found on the classpath. To avoid non-deterministic
-    // loading implementations, please explicitly provide an HTTP client via the client builders, set
-    // the software.amazon.awssdk.http.service.impl system property with the FQCN of the HTTP service to
-    // use as the default, or remove all but one HTTP implementation from the classpath"
-    // Currently, there are 2 implements of HTTP client: ApacheHttpClient and UrlConnectionHttpClient
-    // The UrlConnectionHttpClient is introduced by #16602, and it causes the exception.
-    // So we set the default HTTP client to UrlConnectionHttpClient.
-    // TODO: remove this after we remove ApacheHttpClient
-    private static void setAWSHttpClient() {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:326]: remove it in the iceberg_metadata_reader

File: build.sh [hunk 4]
```diff
@@ -519,6 +519,7 @@ if [ ${BUILD_BE} -eq 1 ]; then
         cp -r -p ${STARROCKS_HOME}/java-extensions/iceberg-metadata-reader/target/iceberg-reader-lib ${STARROCKS_OUTPUT}/be/lib/
         cp -r -p ${STARROCKS_HOME}/java-extensions/iceberg-metadata-reader/target/starrocks-iceberg-metadata-reader.jar ${STARROCKS_OUTPUT}/be/lib/jni-packages
         cp -r -p ${STARROCKS_HOME}/java-extensions/iceberg-metadata-reader/target/starrocks-iceberg-metadata-reader.jar ${STARROCKS_OUTPUT}/be/lib/iceberg-reader-lib
+        cp -r -p ${STARROCKS_HOME}/java-extensions/jni-runtime/target/jni-runtime-lib ${STARROCKS_OUTPUT}/be/lib/
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply build.sh]: fixed

File: java-extensions/jni-runtime/pom.xml (hunk 2 above)
Comment (Smith-Cruise) [reply java-extensions/jni-runtime/pom.xml]: fixed

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java (hunk 3 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:326]: removed

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.3

**Pull request merged**